lib/clients/reports/rates_and_costs_report.ex

```diff
@@ -0,0 +1,66 @@
+defmodule GlassFactoryApi.Clients.Reports.RatesAndCostsReport do
+  @moduledoc """
+  Defines the RatesAndCostsReport of a client report.
+  """
+  alias GlassFactoryApi.Clients.Reports.RatesAndCostsReport
+
+  @type t() :: %RatesAndCostsReport{
+          client_id: integer(),
```

Comment:
Something to consider about type consistency: required and nullable fields.

Our current `defstruct` clause will initialize all fields with `nil`.

```elixir
iex> %RatesAndCostsReport{}
%RatesAndCostsReport{client_id: nil, project_id, nil, …}
```

Therefore, to make this typespec "strictly correct" we'd have to allow for `nil` in some fields, like:

```elixir
client_id: integer | nil,
date: String.t | nil,
# …
```

On the other hand, if we wanted to leave the typespec untouched but enforce this "system invariant" at the same time, we'd probably need to provide "defaults" in `defstruct`:

```elixir
defstruct [
  client_id: 0,
  date: "1900-01-01",
  # …
]
```

What are your thoughts on this?

Reply:
I'm not so confident in setting defaults to this struct because at this moment it's very unclear for me what the API would return, I think that are fields that shouldn't be `nil` but setting a default value might generate some wrong date I think 🤔 

But I do agree that some values are meaningless without others and we can try to define some required fields not just in this struct but in all over the library.

I think we can start defining in this one and spread it across the lib in other PR, what do you guys think?

Reply:
Sure, I've done this change, but I need help of you guys to validate it, because I'm not sure if it is correct 😅:

https://github.com/Cendrao/glass_factory_api/pull/37/commits/94cbaf8b795bdcd246c227fb7b7706006679a5a8

Reply:
@paulo-silva perfect! using `fetch!` makes the code more assertive about the expectations we have on the API response 👍